app/src/hooks/activity-value-form/use-activity-form.ts

```diff
@@ -0,0 +1,78 @@
+import { useForm } from "react-hook-form";
+import { useEffect } from "react";
+import { ActivityValue } from "@/models/ActivityValue";
+import { SuggestedActivity } from "@/util/form-schema";
+import { Inputs } from "@/components/Modals/activity-modal/activity-modal-body";
+
+const useActivityForm = ({
+  targetActivityValue,
+  selectedActivity,
+}: {
+  targetActivityValue: ActivityValue | undefined;
+  selectedActivity?: SuggestedActivity;
+}) => {
+  const {
+    register,
+    handleSubmit,
+    reset,
+    watch,
+    setError,
+    setFocus,
+    formState: { errors },
+  } = useForm<Inputs>();
+
+  useEffect(() => {
+    if (targetActivityValue) {
+      reset({
+        activity: {
+          ...targetActivityValue.activityData,
+          dataQuality: targetActivityValue?.dataSource?.dataQuality,
+          sourceReference: targetActivityValue?.dataSource?.notes,
+          CH4EmissionFactor: targetActivityValue?.activityData?.ch4_amount,
+          CO2EmissionFactor: targetActivityValue?.activityData?.co2_amount,
+          N2OEmissionFactor: targetActivityValue?.activityData?.n2o_amount,
```

Comment:
This seems to be wrong. These need to come from the `emissionsPerActivity` property of the `EmissionsFactor` associated with each `GasValue`.

The fields you're using here are the manual input ones, which don't apply for activity based reporting.